client/layout/masterbar/checkout.tsx

```diff
@@ -73,9 +78,36 @@ const CheckoutMasterbar: FunctionComponent< Props > = ( {
 			console.error( 'Error getting query string in close button' );
 		}
 
+		if ( closeUrl.startsWith( '/' ) ) {
+			page( closeUrl );
+			return;
+		}
 		window.location.href = closeUrl;
```

Comment:
I think you want to use the `navigate( closeUrl )` function from `calypso/lib/navigate`. It will look at the URL and will choose a `page` or browser navigation as appropriate.

And the `checkoutBackUrl` condition above on line 37/42 should merely assign to `closeUrl`. The actual navigation should be done only once at the end.

Reply:
TIL! There's no documentation on `navigate` or on `logmeinUrl` which it calls, but from reading the code it looks like this would cause absolute URLs to be modified with a `logmein=direct` query string? Is that necessary here?

Reply:
Oh, I see now that it only adds that query string for certain categories of sites. Thanks for the info!

Reply:
`navigate` doesn't check if an URL is relative or absolute, but rather whether it's same-origin (`wordpress.com`) or cross-origin. page.js can handle same-origin URL even if it's not in Calypso: it will do `window.location.href =` as a last resort fallback when no route handler is found.

`logmein` is useful when linking to a site with a mapped domain. Then the `logmein` query param will instruct the site to do a redirect loop with remote login. The query param is added only when the URL's domain belongs to one of your sites (looking at Redux `state.sites`) and the site satisfies some extra conditions.